### PR TITLE
Fix variable anchors on globe

### PR DESCRIPTION
**Pull request by mpulkki-mapbox**
Prior to this change labels with variable anchors were placed incorrectly in the globe view. The issue is clearly visible as these labels can be seen placed inside the globe. The fix is to use re-projected positions of label anchors instead of tile position anchors.

One issue remains that is related to the transition logic between mercator and globe projections. We're skipping transition of labels whose placement is computed on the cpu side as the logic is far more complex than the logic of transitioning tile space anchors. This means that for example line aligned labels and labels with variable anchors are not visible during transitions. This is something that should be fixed in a follow-up PR. I believe the shader flag `PROJECTED_POS_ON_VIEWPORT` should be refactored as well in favor of using just `PROJECTION_GLOBE_VIEW`.

bug_globe_variable_anchors

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: src/geo/projection/globe_tile_transform.js [hunk 1]
```diff
@@ -65,9 +66,11 @@ export default class GlobeTileTransform {
         return latLngToECEF(latFromMercatorY(mercY), lngFromMercatorX(mercX), 1.0);
     }
 
-    upVectorScale(id: CanonicalTileID): number {
+    upVectorScale(id: CanonicalTileID, latitude: number, worldSize: number): ElevationScale {
+        const pixelsPerMeterAtLat = mercatorZfromAltitude(1, latitude) * worldSize;
         const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * GLOBE_RADIUS * Math.PI;
-        return pixelsPerMeterECEF * globeECEFNormalizationScale(globeTileBounds(id));
+
+        return {metersToTile: pixelsPerMeterECEF * globeECEFNormalizationScale(globeTileBounds(id)), metersToLabelSpace: pixelsPerMeterAtLat};
```
Comment (karimnaaji) [inline src/geo/projection/globe_tile_transform.js:73]: There might be a lot of redundant calculation of `globeECEFNormalizationScale(globeTileBounds(id))`, which should be the same for every symbols of a tile. Should it be part of the tile transform?

**Review by karimnaaji — APPROVED** [review]
Looks good to me. Left one question about reducing some redundant calculations, which is slightly related to https://github.com/mapbox/mapbox-gl-js/issues/11392.

File: src/geo/projection/globe_tile_transform.js (hunk 1 above)
Comment (karimnaaji) [reply src/geo/projection/globe_tile_transform.js:73]: Similarly `const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * GLOBE_RADIUS * Math.PI;` could be defined as a constant.

Comment (mpulkki-mapbox) [reply src/geo/projection/globe_tile_transform.js:73]: Thanks for the comments! I'll move `pixelsPerMeterECEF` to outside of the scope. We should do a separate refactoring pass to reduce these redundant calculations as you pointed out

**Pull request merged**